### PR TITLE
Add Entry: General: Serialization

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

Add entry from Target Keyword Sheet: General ~Serialization/Deserialization~ Serialization

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by KTom101 — COMMENTED** [review]
@SSwiniarski I've made some minor suggestions. Let me know what you think.

**Comment by SSwiniarski** [comment]
@KTom101, I implemented your suggestions.

File: content/general/concepts/serialization-deserialization/serialization-deserialization.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+Title: 'Serialization/Deserialization'
+Description: 'Serialization is the process of translating a complex object into a format that can be later deserialized back into a copy of the original object.'
```
Comment (yangc95) [inline content/general/concepts/serialization-deserialization/serialization-deserialization.md:3]:
```suggestion
Description: 'Serialization is the process of translating a complex object into a format that can be deserialized back into a copy of the original object.'
```

File: content/general/concepts/serialization-deserialization/serialization-deserialization.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  - 'Classes'
+  - 'Data Structures'
+  - 'JSON'
+  - 'Objects'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+Serialization is the process of translating a complex object into a format, typically a string of characters, that can be later deserialized back into a copy of the original object. This is beneficial as it allows the software to save the state of an arbitrary object to a destination that can be completely unaware of the details of the object being saved. So any object can be saved to a simple string variable, a database field, an external file system, some third-party application in the cloud, or even printed out on a sheet of paper.
```
Comment (yangc95) [inline content/general/concepts/serialization-deserialization/serialization-deserialization.md:17]:
```suggestion
Serialization is the process of translating a complex object into a format, typically a string of characters, that can be deserialized back into a copy of the original object. This is beneficial as it allows the software to save the state of an arbitrary object to a destination that can be completely unaware of its details. Any object can be saved to a simple string variable, database field, external file system, third-party application in the cloud, or even printed out on a sheet of paper.
```

File: content/general/concepts/serialization-deserialization/serialization-deserialization.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+Common formats for serialization include XML and JSON, which have the advantage of being human-readable, but there are also binary formats that aren't human-readable. Regardless of the format, the purpose of serialization is that at a later point in time the serialized data can be read or processed as input and be deserialized back into a copy of the original object.
```
Comment (yangc95) [inline content/general/concepts/serialization-deserialization/serialization-deserialization.md:19]:
```suggestion
Common formats for serialization include XML and JSON, which have the advantage of being human-readable. However, other formats like binary aren't human-readable. Regardless of the format, the purpose of serialization is that at a later point in time, the serialized data can be read or processed as input and be deserialized back into a copy of the original object.
```

**Review by yangc95 — COMMENTED** [review]
1️⃣ done~

**Comment by SSwiniarski** [comment]
@yangc95, I followed your suggestions with a few additional edits of my own, including changing the file name.

File: content/general/concepts/serialization/serialization.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  - 'JSON'
+  - 'Objects'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+Serialization is the process of translating a complex object into a format, typically a string of characters, that can be deserialized back into a copy of the original object. This is beneficial as it allows the software to save the state of an arbitrary object to a destination that can be completely unaware of its details. Any object can be saved to a simple string variable, database field, external file system, third-party application in the cloud, or even printed out on a sheet of paper.
+
+Common formats for serialization include XML and JSON, which have the advantage of being human-readable. However, being human-readable isn't a requirement, and in some cases seriialization use other formats, such as binary, which aren't. Regardless of the format, the purpose of serialization is that at a later point in time, the serialized data can be read or processed as input and be deserialized back into a copy of the original object.
```
Comment (yangc95) [inline content/general/concepts/serialization/serialization.md:19]:
```suggestion
Common formats for serialization include XML and JSON, which have the advantage of being human-readable. However, that is not a requirement, and in some cases, serialization uses other formats such as binary which aren't human-readable. Regardless of the format, the purpose of serialization is that at a later point in time, the serialized data can be read or processed as input and be deserialized back into a copy of the original object.
```

File: content/general/concepts/serialization/serialization.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+When programming languages implement serialization, they often require special declarations in defined classes to make their associated objects serializable, and often require special libraries to implement this functionality.
```
Comment (yangc95) [inline content/general/concepts/serialization/serialization.md:21]:
```suggestion
When programming languages implement serialization, they often require special declarations in defined classes to make their associated objects serializable and often require special libraries to implement this functionality.
```

**Review by yangc95 — COMMENTED** [review]
Just a few small points and I believe it's ready to merge! @KTom101 any more changes before we do so? 🙂

**Comment by KTom101** [comment]
@yangc95 I think once your changes are made, it's all good to merge. I don't see any more changes on my end.

**Comment by SSwiniarski** [comment]
@yangc95: Changes have been implemented.

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
@SSwiniarski Live link: https://www.codecademy.com/resources/docs/general/serialization

Ty @KTom101 !

**Pull request merged**